### PR TITLE
NO-JIRA: Record Kube event first/last timestamp in the resulting intervals

### PR DESCRIPTION
**Pull request by dgoodwin**

### CODE REVIEW

**Comment by dgoodwin** [comment]
Updated with a fix for the unit tests.

File: pkg/monitortests/testframework/watchevents/event.go [hunk 1]
```diff
@@ -172,6 +172,10 @@ func recordAddOrUpdateEvent(
 		}
 		return
 	}
+
+	message = message.WithAnnotation("FirstTimestamp", obj.FirstTimestamp.Format(time.RFC3339))
```
Comment (eggfoobar) [inline pkg/monitortests/testframework/watchevents/event.go:176]: Small nit, was looking at the output data seems like the other annotations are in camel case, should we change this to that as well?

```json
                "annotations": {
                    "FirstTimestamp": "2024-04-20T05:35:19Z",
                    "LastTimestamp": "2024-04-20T06:00:28Z",
                    "count": "24",
                    "interesting": "true",
                    "pathological": "true",
                    "reason": "EtcdCertSignerControllerUpdatingStatus"
                }
```

**Comment by dgoodwin** [comment]
Good catch, will update.

**Comment by dgoodwin** [comment]
Updated, just needs an lgtm.

**Pull request merged**